test/contracts/kill_switch/issues_registry.test.js

```diff
@@ -0,0 +1,129 @@
+const { SEVERITY } = require('./enums')
+const { assertRevert } = require('../../helpers/assertThrow')
+const { assertAmountOfEvents, assertEvent } = require('../../helpers/assertEvent')(web3)
+const { getEventArgument, getNewProxyAddress } = require('../../helpers/events')
+
+const IssuesRegistry = artifacts.require('IssuesRegistry')
+const ACL = artifacts.require('ACL')
+const Kernel = artifacts.require('Kernel')
+const KillSwitch = artifacts.require('KillSwitch')
+const DAOFactory = artifacts.require('DAOFactory')
+const EVMScriptRegistryFactory = artifacts.require('EVMScriptRegistryFactory')
+
+contract('IssuesRegistry', ([_, root, implementation, owner, anyone]) => {
+  let kernelBase, aclBase, issuesRegistryBase, registryFactory, dao, acl, issuesRegistry, killSwitchBase
+
+  before('deploy base implementations', async () => {
+    kernelBase = await Kernel.new(true) // petrify immediately
+    aclBase = await ACL.new()
+    killSwitchBase = await KillSwitch.new()
+    registryFactory = await EVMScriptRegistryFactory.new()
+    issuesRegistryBase = await IssuesRegistry.new()
+  })
+
+  before('deploy DAO', async () => {
+    const daoFactory = await DAOFactory.new(kernelBase.address, aclBase.address, killSwitchBase.address, registryFactory.address)
+    const kernelReceipt = await daoFactory.newDAO(root)
+    dao = Kernel.at(getEventArgument(kernelReceipt, 'DeployDAO', 'dao'))
+    acl = ACL.at(await dao.acl())
+    const APP_MANAGER_ROLE = await kernelBase.APP_MANAGER_ROLE()
+    await acl.createPermission(root, dao.address, APP_MANAGER_ROLE, root, { from: root })
+  })
+
+  beforeEach('create issues registry', async () => {
+    const receipt = await dao.newAppInstance('0x1234', issuesRegistryBase.address, '0x', false, { from: root })
+    issuesRegistry = IssuesRegistry.at(getNewProxyAddress(receipt))
+    await issuesRegistry.initialize()
+    const SET_SEVERITY_ROLE = await issuesRegistryBase.SET_SEVERITY_ROLE()
+    await acl.createPermission(owner, issuesRegistry.address, SET_SEVERITY_ROLE, root, { from: root })
+  })
+
+  describe('hasSeverity', () => {
+    context('when there was no severity set before', () => {
+      it('returns false', async () => {
+        assert.isFalse(await issuesRegistry.hasSeverity(implementation), 'did not expect severity for given implementation')
+      })
+    })
+
+    context('when there was a severity already set', () => {
+      beforeEach('set medium severity', async () => {
+        await issuesRegistry.setSeverityFor(implementation, SEVERITY.LOW, { from: owner })
+      })
+
+      context('when the issues was not fixed yet', () => {
+        it('returns true', async () => {
+          assert.isTrue(await issuesRegistry.hasSeverity(implementation), 'did not expect severity for given implementation')
+        })
+      })
+
+      context('when the issues was already fixed', () => {
```

Comment:
```suggestion
      context('when the issues was a false positive', () => {
```

It's unrealistic that the issue would be fixed without updating the base implementation contract (it would have to be an dynamically linked external contract that got changed), so this might be a more realistic scenario 😄